dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/DialogueRequestAnnotationsProcessor.java

```diff
@@ -105,11 +104,14 @@ public boolean process(Set<? extends TypeElement> _annotations, RoundEnvironment
                     }
 
                     try {
-                        Goethe.formatAndEmit(javaFile, filer, interfaceElement);
-                    } catch (FilerException e) {
-                        // Happens when same file is written twice. Dunno why this is a problem
-                    } catch (IOException e) {
-                        error("Failed to format generated code", interfaceElement, e);
+                        Goethe.formatAndEmit(javaFile, filer);
+                    } catch (RuntimeException e) {
+                        if (e.getCause() instanceof FilerException) {
+                            // Happens when same file is written twice.
+                            // This indicates additional data was discovered in a subsequent processing round.
```

Comment:
@jkozlowski I updated the comment here, is this a case we need to handle?

Reply:
Honestly I don't remember. Let's merge and FLUP if you want. If other processors are not running into this, it's probably fine.